src/python/WMCore/ReqMgr/CherryPyThreads/HeartbeatMonitor.py

```diff
@@ -1,5 +1,186 @@
+        return ["NONE"]
+    elif isinstance(propertyValue, basestring):
+        return [propertyValue]
+    # then it's a list
+    return propertyValue
+
+
+def _getRequestNumEvents(propertyValue):
+    """
+    WMStats server can return this argument in three different data types.
+    If there are multiple values, return only the first one.
+    """
+    if not propertyValue:
+        return 0
+    elif isinstance(propertyValue, list):
+        return propertyValue[0]
+    # then it's integer
+    return propertyValue
+
+
+def initMetrics(veryActiveStatus):
+    """
+    Initialize the metrics dictionary
+    :param veryActiveStatus: a small list of active status
+    """
+    results = {"requestsByStatus": {},
+               "requestsByStatusAndCampaign": {},
+               "requestsByStatusAndPrio": {},
+               "requestsByStatusAndNumEvts": {}}
+
+    for st in ACTIVE_STATUS:
+        results["requestsByStatus"].setdefault(st, 0)
+        results["requestsByStatusAndCampaign"].setdefault(st, {})
+
+    for st in veryActiveStatus:
+        results["requestsByStatusAndPrio"].setdefault(st, {})
+        results["requestsByStatusAndNumEvts"].setdefault(st, 0)
+
+    return results
+
 
 class HeartbeatMonitor(HeartbeatMonitorBase):
-    pass
+
+    def addAdditionalMonitorReport(self, config):
+        """
+        _addAdditionalMonitorReport_
+
+        Collect general request information and post it to both central
+        couchdb and MonIT. Items to fetch:
+         * number of requests in each status (excluding archived ones)
+         * for requests in assignment-approved:
+           * their priority
+           * their RequestNumEvents (if there is no input dataset)
+         * for assignment-approved, assigned, acquired, running-* and completed:
+           * number of requests in each campaign
+        """
+        self.logger.info("Collecting ReqMgr2 statistics...")
+        VERY_ACTIVE_STATUS = ["assignment-approved", "acquired",
+                              "running-open", "running-closed"]
+        wmstatsSvc = WMStatsServer(config.wmstatsSvc_url, logger=self.logger)
+
+        results = initMetrics(VERY_ACTIVE_STATUS)
+
+        inputConditon = {}
+        outputMask = ["RequestStatus", "RequestType", "RequestPriority",
+                      "Campaign", "RequestNumEvents"]
+        startT = int(time.time())
+        for reqInfo in wmstatsSvc.getFilteredActiveData(inputConditon, outputMask):
+            status = reqInfo['RequestStatus']
+            results['requestsByStatus'][status] += 1
+
+            for campaign in _getCampaign(reqInfo["Campaign"]):
+                results["requestsByStatusAndCampaign"][status].setdefault(campaign, 0)
+                results['requestsByStatusAndCampaign'][status][campaign] += 1
+
+            if status not in VERY_ACTIVE_STATUS:
+                continue
+
+            requestPrio = reqInfo['RequestPriority']
```

Comment:
So this one is for all the active status not just VERY_ACTIVE_STATUS, right?

Reply:
Exactly, though I'm not 100% sure. Still, I think people don't care about request priority for requests in completed, or aborted, or ... closed-out etc, I think...

Reply:
I wonder whether it will be more confusing select certain state for certain condition but not other conditions, we can report everything but it doesn't have to be used.

Reply:
Yes, I slept on that and I think it's better to don't use a small set of ACTIVE_STATUS, but fetch the same metrics for all of them. Thanks! I'm updating this PR